xarray/tests/test_dataset.py

```diff
@@ -1181,6 +1557,24 @@ def test_reindex(self):
         actual = ds.reindex(x=[0, 1, 3], y=[0, 1])
         self.assertDatasetIdentical(expected, actual)
 
+    def test_reindex_warning(self):
+        data = create_test_data()
+
+        with pytest.warns(FutureWarning) as ws:
+            # DataArray with different dimension raises Future warning
+            ind = xr.DataArray([0.0, 1.0], dims=['new_dim'], name='ind')
+            data.reindex(dim2=ind)
+            assert any(["Indexer has dimensions " in
+                        str(w.message) for w in ws])
+
+        with pytest.warns(FutureWarning) as ws:
+            # Should not warn
+            ind = xr.DataArray([0.0, 1.0], dims=['dim2'], name='ind')
+            data.reindex(dim2=ind)
+            assert all(["Indexer has dimensions " not in
+                        str(w.message) for w in ws])
+            warnings.warn('dummy', FutureWarning, stacklevel=3)
+
```

Comment:
2 things here:

1) I'm not sure why we need the dummy warning if it should not warn. Can we assert that a warning was not raised?
2) move the `ind = ...` line out of the context manager so it is clear which line should be raising a warning

Reply:
I do not know a good way to make sure that the code does NOT warn.
If the dummy warning line is removed, the test fails because the code does not warn.

Does anyone know a better workaround?

Reply:
I think this [SO](https://stackoverflow.com/questions/45671803/how-to-use-pytest-to-assert-no-warning-is-raised) answer should work.

```Python
ind = xr.DataArray([0.0, 1.0], dims=['dim2'], name='ind')
with pytest.warns(None) as ws:
    # Should not warn
    data.reindex(dim2=ind)
    assert len(ws) == 0

Reply:
Thanks. It now works without dummy warning.